ndmg/scripts/ndmg_bids.py

```diff
@@ -447,16 +453,18 @@ def main():
     # it's super gross.
     if buck is not None and remo is not None:
         if subj is not None:
-            if len(sesh) == 1:
-                sesh = sesh[0]
+            #if len(sesh) == 1:
+            #    sesh = sesh[0]
             for sub in subj:
                 if sesh is not None:
-                    remo = op.join(remo, "sub-{}".format(sub), "ses-{}".format(sesh))
-                    tindir = op.join(inDir, "sub-{}".format(sub), "ses-{}".format(sesh))
+                    for ses in sesh:
+                        rem = op.join(remo, "sub-{}".format(sub), "ses-{}".format(ses))
+                        tindir = op.join(inDir, "sub-{}".format(sub), "ses-{}".format(ses))
+                        s3_utils.s3_get_data(buck, rem, tindir, public=not creds)
                 else:
-                    remo = op.join(remo, "sub-{}".format(sub))
+                    rem = op.join(remo, "sub-{}".format(sub))
                     tindir = op.join(inDir, "sub-{}".format(sub))
-                s3_utils.s3_get_data(buck, remo, tindir, public=not creds)
+                    s3_utils.s3_get_data(buck, rem, tindir, public=not creds)
```

Comment:
Could you briefly exchange these changes and the reasoning? I think I understand what's going on here, but would need to hop into a debugger and play with it a bit to know for sure

Make sure you've tested this with `s3` and everything, since it's pretty easy to break the pipeline changing this block of code

on a side note, I hate how messy this code block is, lol

Reply:
Yeah, so before if you didn't specify the sessions, it tries to find the length and throws an Error. The other changes are to  fix the fact that if you have more than one session specified it tries to find a non-existent file. 
For example, Subject = 245, Sessions = 1, 2
It will try to find the directory  .../sub-245/ses-[1 2]/
This was the error I was talking about yesterday and this change I made works

Reply:
Ok, sweet. What have you tested this with to verify?
(e.g., local input folder, 1 subject, 1 session, local input folder, 2 subjects, 2 sessions, s3, etc)

Reply:
It doesn't effect local folders, as this code is only used when the user specifies a bucket. I've tested it with single and multiple subject/session combinations. Though it does not work if the user specifies sessions without subjects, it will just default to getting all subjects and all sessions in the s3 location

Reply:
Oh yeah, woops, you're right! Forgot about that top-level line.
Great, good job! Resolved.